internal/pkg/deploy/upload/asset/asset_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package asset
+
+import (
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Upload(t *testing.T) {
+	const mockContent = "mockContent"
+	testCases := map[string]struct {
+		inSource       string
+		inIncludes     []string
+		inExcludes     []string
+		mockFileSystem func(fs afero.Fs)
+
+		expectedFiles []string
+		expectedError error
+	}{
+		"success without include and exclude": {
+			inSource: "./test",
+			mockFileSystem: func(fs afero.Fs) {
+				afero.WriteFile(fs, "test/copilot/.workspace", []byte(mockContent), 0644)
+				afero.WriteFile(fs, "copilot/prod/manifest.yaml", []byte(mockContent), 0644)
+			},
+			expectedFiles: []string{"test/copilot/.workspace"},
+		},
+		"success with include only": {
+			inSource:   "./test",
+			inIncludes: []string{"copilot/prod/manifest.yaml"},
+			mockFileSystem: func(fs afero.Fs) {
+				afero.WriteFile(fs, "test/copilot/.workspace", []byte(mockContent), 0644)
+				afero.WriteFile(fs, "copilot/prod/manifest.yaml", []byte(mockContent), 0644)
+			},
+			expectedFiles: []string{"test/copilot/.workspace"},
```

Comment:
+1 to having comment doc besides `Includes` and `Excludes`. I thought the `expectedFiles` here should be `[]string{"test/copilot/.workspace"}` because I didn't understand what `Includes` was.